app/assets/javascripts/components/overview/statistics_update_info.jsx

```diff
@@ -19,27 +19,47 @@ const StatisticsUpdateInfo = createReactClass({
 
   getUpdateTimesInformation() {
     const course = this.props.course;
-    const updateTimesLogs = Object.values(course.flags.update_logs).filter(log => log.end_time !== undefined);
-
-    if (updateTimesLogs.length === 0) return null;
-
-    const lastSuccessfulUpdate = updateTimesLogs[updateTimesLogs.length - 1].end_time;
-    const lastSuccessfulUpdateMoment = moment.utc(lastSuccessfulUpdate);
-    const averageDelay = course.updates.average_delay;
     let lastSuccessfulUpdateMessage = '';
     let nextUpdateMessage = '';
-    let isNextUpdateAfter = null;
-
-    if (lastSuccessfulUpdate) {
-      lastSuccessfulUpdateMessage = `${I18n.t('metrics.last_update')}: ${lastSuccessfulUpdateMoment.fromNow()}`;
-      const nextUpdateExpectedTime = lastSuccessfulUpdateMoment.add(averageDelay, 'seconds');
+    let isNextUpdateAfter = false;
+    if (course.updates.last_update === null) {
+     lastSuccessfulUpdateMessage = `${I18n.t('metrics.no_update')}`;
+     if (course.flags.first_update) {
+      const nextUpdateExpectedTime = this.firstUpdateTime();
       isNextUpdateAfter = nextUpdateExpectedTime.isAfter();
-      nextUpdateMessage = `${I18n.t('metrics.next_update')}: ${nextUpdateExpectedTime.fromNow()}`;
-    }
+      nextUpdateMessage = `${I18n.t('metrics.first_update')}: ${nextUpdateExpectedTime.fromNow()}`;
+      lastSuccessfulUpdateMessage = `${I18n.t('metrics.enqueued_update')}`;
+     }
+     } else {
+      const lastSuccessfulUpdate = this.lastSuccessfulUpdateTime();
+      const lastSuccessfulUpdateMoment = moment.utc(lastSuccessfulUpdate);
+      const averageDelay = course.updates.average_delay;
 
+      if (lastSuccessfulUpdate) {
+        lastSuccessfulUpdateMessage = `${I18n.t('metrics.last_update')}: ${lastSuccessfulUpdateMoment.fromNow()}`;
+        const nextUpdateExpectedTime = lastSuccessfulUpdateMoment.add(averageDelay, 'seconds');
+        isNextUpdateAfter = nextUpdateExpectedTime.isAfter();
+        nextUpdateMessage = `${I18n.t('metrics.next_update')}: ${nextUpdateExpectedTime.fromNow()}`;
+      }
+    }
     return [lastSuccessfulUpdateMessage, nextUpdateMessage, isNextUpdateAfter];
   },
 
+  firstUpdateTime() {
```

Comment:
I suggest isolating these even further. Neither of these functions has any side effects in terms of React, so there's no need for them to be part of the component if they take the `course` object as their argument. I recommend either moving them to a separate file entirely (as you did previously with  wiki_utils.js) or moving them out of the component as standalone `const` functions outside of the React class (which you can see an example of in `alerts_handler.jsx`).